File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ret;
        TreeNode *curr = NULL;
        int size = 0;
        int index = 0;
        queue<TreeNode*> q;
        
        if (root)
            q.push(root);
        while (!q.empty()) {
            size = q.size();
            ret.resize(index + 1);
            for (int i=0; i<size; ++i) {
                curr = q.front();
                q.pop();
                ret[index].push_back(curr->val);
                if (curr->left) 
                    q.push(curr->left);
                if (curr->right) 
                    q.push(curr->right);
            }
            ++index;
        }
        
        return ret;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []
        
        res = []
        queue = []
        
        queue.append(root)
        while queue:
            curr = []
            for i in range(len(queue)):
                tmp = queue.pop(0)
                curr.append(tmp.val)
                if tmp.left != None:
                    queue.append(tmp.left)
                if tmp.right != None:
                    queue.append(tmp.right)
            res.append(curr)
            
        return res